packages/hydrooj/src/service/server.ts

```
@@ -73,10 +75,9 @@ export function requireSudo(target: any, funcName: string, obj: any) {
 
 export interface Handler {
     domain: DomainDoc;
-    ctx: Context;
 }
-export class Handler extends HandlerOriginal {
-    constructor(_, ctx: Context) {
+export class Handler extends HandlerOriginal<Context> {
```

Biome (1.9.4)

[error] This class is unsafely merged with an interface.

The interface is declared here.

The TypeScript compiler doesn't check whether properties defined in the interface are initialized in the class.

(lint/suspicious/noUnsafeDeclarationMerging)